packages/@react-aria/virtualizer/src/VirtualizerItem.tsx

```diff
@@ -57,8 +57,12 @@ export function layoutInfoToStyle(layoutInfo: LayoutInfo, dir: Direction, parent
   }
 
   let rectStyles = {
-    top: layoutInfo.rect.y - (parent ? parent.rect.y : 0),
-    [xProperty]: layoutInfo.rect.x - (parent ? parent.rect.x : 0),
+    // TODO: For layoutInfos that are sticky that have parents with overflow visible, their "top" will be relative to the to the nearest scrolling container
+    // which WON'T be the parent since the parent has overflow visible. This means we shouldn't offset the height by the parent's position
+    // Not 100% about this change here since it is quite ambigious what the scrolling container maybe and how its top is positioned with respect to the
+    // calculated layoutInfo.y here
+    top: layoutInfo.rect.y - (parent && !(parent.allowOverflow && layoutInfo.isSticky) ? parent.rect.y : 0),
+    [xProperty]: layoutInfo.rect.x - (parent && !(parent.allowOverflow && layoutInfo.isSticky) ? parent.rect.x : 0),
```

Comment:
Checked that this doesn't break things in RSP TableView but still might be a bit of a tricky change due to how sticky positioned elements calculate their position

Reply:
Is this assuming that an element with allowsOverflow and isSticky also won't have position relative or absolute? Not sure we can assume that

Reply:
well, `isSticky` would mean the position is set to `sticky` rather than `relative` or `absolute` no?

Reply:
This isn't quite right because it is relative to the nearest scrolling or `overflow: hidden` container, which might not be the root. So really `parent` here should be either the immediate parent in case of `position: absolute`, or the nearest ancestor without `allowOverflow` if `position: sticky`. In the table case that doesn't happen because all of the ancestors have `allowOverflow` set, but it could in theory.

Alternatively, if we used `overflow: clip` instead of `overflow: hidden` the sticky elements would always be positioned relative to the root like you are assuming here. Looks like I tried that in https://github.com/adobe/react-spectrum/pull/6613 but reverted it due to some other side effects. We may be able to revisit that at some point.

For now I'd like to ship this as is since we've already tested it, and follow up later with a more correct solution.